### PR TITLE
Clean up options and fix test code

### PR DESCRIPTION
**Pull request by echo304**
- Clean up options and create new `prepack-options.js` file
- Add `PartialEvaluatorOptions` type which contains only `sourceMaps`
- Fix test code related to above change

https://github.com/facebook/prepack/issues/841

### CODE REVIEW

File: src/prepack-options.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  uniqueSuffix,
+  residual,
+  serialize = !residual,
+  strictlyMonotonicDateNow,
+  timeout,
+}: Options): RealmOptions {
+  return {
+    compatibility,
+    debugNames,
+    errorHandler: errorHandler,
```
Comment (NTillmann) [inline src/prepack-options.js]: simplify to `errorHandler,`

File: src/prepack-options.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  logModules = false,
+  profile = false,
+  singlePass = false,
+  speculate = false,
+  trace = false,
+}: Options): SerializerOptions {
+  let result: SerializerOptions = {
+    delayInitializations,
+    delayUnsupportedRequires,
+    initializeMoreModules: speculate,
```
Comment (NTillmann) [inline src/prepack-options.js]: Another silly naming discrepancy that should be addressed eventually. If that's harmonized, then the `Options` datatype could be truly the union `SerializerOptions | RealmOptions | PartialEvaluatorOptions`, right?

File: src/prepack-options.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+/* @flow */
+
+import type { ErrorHandler } from "./errors.js";
+import type { SerializerOptions, RealmOptions } from "./options";
+
+export type Compatibility = "browser" | "jsc-600-1-4-17" | "node-source-maps" | "node-cli";
+export const CompatibilityValues = ["browser", "jsc-600-1-4-17", "node-source-maps", "node-cli"];
+
+export type Options = {|
```
Comment (NTillmann) [inline src/prepack-options.js]: You could rename this type to `PrepackOptions`, or `CombinedPrepackOptions`, to make it a bit more clear what this type represents.

Comment (echo304) [inline src/prepack-options.js]: let's pick `PrepackOptions`

File: src/prepack-options.js (hunk 2 above)
Comment (echo304) [inline src/prepack-options.js]: good call 👍

**Comment by NTillmann** [comment]
Awesome! @cblappert, do you think this might break any internal tooling?

**Pull request closed without merging**